### PR TITLE
Add helper to sync maintainers file

### PR DESCRIPTION
**Pull request by jepio**

### CODE REVIEW

File: sync-maintainers/sync-maintainers.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+{paragraph}
+
+The contents of this file are synchronized from [Flatcar/MAINTAINERS.md](https://github.com/flatcar-linux/Flatcar/blob/main/MAINTAINERS.md).
+'''
+
+def write_maintainers_file(repo_name, paragraph, maintainers):
+    maintainers_entry = '\n'.join(maintainers)
+    maintainers_content = MAINTAINERS_TEMPLATE.format(maintainers=maintainers_entry, paragraph=paragraph)
+    repo_filename = '{}/MAINTAINERS.md'.format(repo_name)
```
Comment (sayanchowdhury) [inline sync-maintainers/sync-maintainers.py]:
```suggestion
    repo_filename = f'{repo_name}/MAINTAINERS.md'
```

File: sync-maintainers/sync-maintainers.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    if not token:
+        raise Exception('Missing GITHUB_TOKEN env variable')
+    repos, _ = parse_maintainers(args.repo)
+    for (repo_name, maintainers) in repos:
+        pr = get_pr(repo_name, token).json()
+        if not pr:
+            print('{} creating pr'.format(repo_name))
+            base = get_default_branch(repo_name, token)
+            pr = create_pr(repo_name, token, base).json()
+            pr = [pr]
```
Comment (sayanchowdhury) [inline sync-maintainers/sync-maintainers.py]:
```suggestion
            pr = [create_pr(repo_name, token, base).json()]
```

File: sync-maintainers/sync-maintainers.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+parser_github = subparser.add_parser('github', help='perform github pull request operations')
+parser_github.add_argument('--repo', help='Repository to operate on; default all')
+parser_github.set_defaults(func=main_github)
+parser_list = subparser.add_parser('list', help='list all repositories with entries')
+parser_list.set_defaults(func=main_list)
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+    args.func(args)
+    #main()
```
Comment (sayanchowdhury) [inline sync-maintainers/sync-maintainers.py]:
```suggestion
```

**Review by sayanchowdhury — COMMENTED** [review]
Looks good. All the `.format` could be changed to f-strings and would be great if you can run [black](https://pypi.org/project/black/) to format the code.

File: sync-maintainers/sync-maintainers.py (hunk 1 above)
Comment (jepio) [reply sync-maintainers/sync-maintainers.py]: replaced all format calls (except one) with f-strings

**Comment by jepio** [comment]
Ran black and replaced format with f-strings (forgot those exist 😄)

**Review by sayanchowdhury — APPROVED** [review]

File: sync-maintainers/sync-maintainers.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    return para, repos
+
+
+MAINTAINERS_TEMPLATE = """# Maintainers
+
+{maintainers}
+
+{paragraph}
+
+The contents of this file are synchronized from [Flatcar/MAINTAINERS.md](https://github.com/flatcar-linux/Flatcar/blob/main/MAINTAINERS.md).
```
Comment (krnowak) [inline sync-maintainers/sync-maintainers.py]:
```suggestion
The contents of this file are synchronized from [Flatcar/MAINTAINERS.md](https://github.com/flatcar/Flatcar/blob/main/MAINTAINERS.md).
```

**Review by krnowak — APPROVED** [review]

**Pull request merged**